### PR TITLE
Add PoC fault injection in fake installer

### PR DESCRIPTION
**Pull request by lbonn**
Requires libfiu-dev on Debian derived distros.

Then:

```
cmake .. -DFAULT_INJECTION=on
make aktualizr
fiu-run -c 'enable name=fake_package_install' aktualizr -c . once
```

### CODE REVIEW

File: cmake-modules/FindLibfiu.cmake [hunk 1]
```diff
@@ -0,0 +1,19 @@
+if (LIBFIU_LIBRARY AND LIBFIU_INCLUDE_DIR)
+  # in cache already
+  set(LIBFIU_FOUND TRUE)
+else (LIBFIU_LIBRARY AND LIBFIU_INCLUDE_DIR)
+  find_package(PkgConfig QUIET)
+  if (PKG_CONFIG_FOUND)
+      pkg_check_modules(systemd_PKG QUIET fiu)
```
Comment (lbonn) [inline cmake-modules/FindLibfiu.cmake]: oops I made some copy-paste errors here

**Comment by lbonn** [comment]
@patrickvacek I've added a doc to explain how to run it. I don't know if unit tests are relevant in the current state but I imagine that this library will be handy for writing more complete tests and that we'll quickly enable faults on CI for these (also mentioned in the doc).

In the meantime, would be nice if someone would go through the guide and reproduce the steps :).

**Comment by patrickvacek** [comment]
So this means we can close OTA-809 (the spike to figure out how to do installation failure), right?

File: docs/fault-injection.adoc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+== Setup
+
+libfiu needs to be installed on the target machine. For local compilation on a Debian-derived system, the distribution package can be used:
+
+    apt install fiu-utils libfiu-dev
+
+Fault injection must then be enabled at CMake configure time with the `-DFAULT_INJECTION=on` option (refer to general building instructions for more details).
+
+`fi-run` and `fi-ctrl` can now be used on the newly compiled aktualizr binary to inject faults (refer to corresponding man pages).
```
Comment (stevana) [inline docs/fault-injection.adoc]:
```suggestion
`fiu-run` and `fiu-ctrl` can now be used on the newly compiled aktualizr binary to inject faults (refer to corresponding man pages).
```

File: src/libaktualizr/utilities/CMakeLists.txt [hunk 3]
```diff
@@ -7,6 +7,7 @@ set(SOURCES dequeue_buffer.cc
 set(HEADERS config_utils.h
             dequeue_buffer.h
             exceptions.h
+            fiu-local.h
```
[the remark this answers is not in the document]
Comment (lbonn) [reply src/libaktualizr/utilities/CMakeLists.txt]: I kept the original file name, but there is no real reason not to change it. Will fix.

File: docs/fault-injection.adoc (hunk 2 above)
Comment (lbonn) [reply docs/fault-injection.adoc]: Thanks!

**Comment by lbonn** [comment]
@patrickvacek  The linking error should be fixed now

**Pull request merged**